File: LeetCode/Java/0771. Jewels and Stones #1.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character, Integer> counter = new HashMap<>();
        int count = 0;
        
        for (int i = 0; i < stones.length(); i++) {
            char stone = stones.charAt(i);
            counter.put(stone, counter.getOrDefault(stone, 0) + 1);
        }
        
        for (int i = 0; i < jewels.length(); i++) {
            char jewel = jewels.charAt(i);
            count += counter.getOrDefault(jewel, 0);
        }
        
        return count;
    }
}

File: LeetCode/Python/0771. Jewels and Stones #1.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        from collections import Counter
        
        stones = Counter(stones)
        count = 0
        
        for jewel in jewels:
            count += stones.get(jewel, 0)
        
        return count
